packages/react-components/src/lib/ol-map/map.tsx

```diff
@@ -22,17 +30,37 @@ export const useMap = (): OlMap => {
   return map;
 };
 
-export const Map: React.FC = (props) => {
+export const Map: React.FC<MapProps> = (props) => {
   const CENTER_LAT = 35,
-    CENTER_LON = 32;
+    CENTER_LON = 32,
+    PROJECTION = 'EPSG:4326',
+    DEFAULT_ZOOM = 10,
+    COORDINATES_FRACTION_DIFITS = 5;
   const mapElementRef = useRef<HTMLDivElement>(null);
+  const {allowFullScreen, showMousePosition} = props;
+  const mapControlExtends = [];
```

Comment:
take a look, runs when props changes and not on each render

Reply:
i added code to log the controls array before and after the controls are added to the array.
![image](https://user-images.githubusercontent.com/12687466/91041724-2f476380-e619-11ea-840f-cf837264c0dc.png)

every time a render happens the controls are created and added into the array for no reason.

![map controls](https://user-images.githubusercontent.com/12687466/91042030-c1e80280-e619-11ea-8da1-f52c610b33ea.gif)

Reply:
Yep, I understand the issue, you still with "problematic code". You can pull/review new version(pushed today).
In addition not sure that this is and indication. React more "clever" with DOM updates, in your case you can see that actual DOM isn't updated. Anyhow move code to useEffect(), now it's also sensitive to property change when map component "lives"

Reply:
my only problem is that the constructors are running each time and creating objects that immediately get garbage collected. nothing to do with the DOM. if you don't mind that ill ignore it.

Reply:
In general, it might be Ok (relay on React in this case). But anyhow you are talking about **OLD code**.
I **already** perform a fix in this commit https://github.com/MapColonies/shared-components/pull/34/commits/87da3493d0db10989fccd02e093b222625a35283 looks better and "sensitive" to props change as well(that what you want)